core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -450,11 +445,11 @@ private boolean isEndpointGatewayEnabled(EnvironmentNetworkResponse network) {
     }
 
     private Optional<InstanceGroup> getOozieHAInstanceGroup(Stack stack) {
-        if (stack.getStackVersion() != null &&
-            isVersionNewerOrEqualThanLimited(stack.getStackVersion(), CLOUDERA_STACK_VERSION_7_2_11)) {
-            Cluster cluster = stack.getCluster();
-            if (cluster != null) {
-                CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(cluster.getBlueprint().getBlueprintText());
+        Cluster cluster = stack.getCluster();
+        if (cluster != null) {
+            CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(cluster.getBlueprint().getBlueprintText());
+            String cdhVersion = cmTemplateProcessor.getStackVersion();
+            if (cdhVersion != null && isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_11)) {
```

Comment:
Great question. In my testing, I found that if we check the Stack object's version, it comes from the Image object, which makes the check invalid because the patch version is attached to it and kicks in the logic for even upgraded clusters. So using the template processor's version so that only the newly created clusters get the LB logic.

Reply:
So, when this was running against clusters that had been upgraded to version 7.2.11+, we were looking up Oozie instance groups.

Checking the CDH version from the template processor lets us use the `"cdhVersion": "7.2.11",` line in`*.bp` files to determine if we should be retrieving Oozie instance groups?

Reply:
Yes, that is correct.